components/script/dom/idbindex.rs

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+use dom_struct::dom_struct;
+use script_bindings::codegen::GenericBindings::IDBIndexBinding::IDBIndexMethods;
+use script_bindings::str::DOMString;
+
+use crate::dom::bindings::reflector::{Reflector, reflect_dom_object};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::idbobjectstore::IDBObjectStore;
+use crate::script_runtime::CanGc;
+
+#[dom_struct]
+pub struct IDBIndex {
```

Comment:
`pub(crate)` for all new code, please! I'm planning to do another sweep through the script crate to clean up a bunch of `pub` uses that have crept in recently.